### PR TITLE
Add collaborators management views

### PR DESCRIPTION
**Pull request by artonge**
Also contains some minor css tweaks.

## Screenshots

Collaborator list | With some selection | Without public link during creation
-- | -- | --
Screenshot from 2022-08-25 12-46-44 | Screenshot from 2022-08-25 12-46-53 |  Screenshot from 2022-08-25 13-30-38

Initial loading | Loading when searching
-- | --
Screenshot from 2022-08-25 12-46-42 | Screenshot from 2022-08-25 12-47-04

Public link created | Public link copied
-- | --
Screenshot from 2022-08-25 12-46-16 | Screenshot from 2022-08-25 12-46-18

Fix https://github.com/nextcloud/photos/issues/1226#

### CODE REVIEW

**Review by jancborchardt — CHANGES_REQUESTED** [review]
Looks good, just a few things:
- [ ] The "Share link" finction looks oddly placed and not interactive. Maybe better as a secondary button on the bottom left instead?
- [ ] The gab below the input field looks off. Wouldn't this work like in Files, where people are picked one by one from the suggestion dropdown?
- [ ] Possibly a general Vue component thing, but there should be a close x within the modal on the top right
- [ ] Wording in the title subline: "users" to "people", and " group" needs to be plural "groups"

**Comment by nimishavijay** [comment]
Looks very nice! Agree with everything mentioned, plus a few more points:
- "Save collaborators list" --> "Save" or maybe "Save collaborators"
- Usually the icon associated with "Save" is the checkmark icon. The paper plane icon is usually for send. I'm also wondering if we need an icon at all 
- "Share this album via link" --> "Share via public link"? To make it clearer that it's a public link and to make it shorter if it's going to be a button on the bottom left
- We can also add an icon to the "Share via public link" button as when it's clicked it changes into the "Copy link" button which has an icon 
- Maybe there can also be a way to unshare in case the "Share via public link" button was clicked by accident. If we are going with the secondary button on the bottom left, then there could possibly be:
  - a trash icon or close icon next to it on the right which would unshare
  - or a 3 dot menu with one item "Unshare"

cc @jancborchardt as well

**Comment by skjnldsv** [comment]
Please rebase after #1228

**Review by skjnldsv — APPROVED** [review]

File: lib/Controller/PreviewController.php [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+use OCP\Files\Folder;
+use OCP\Files\IRootFolder;
+use OCP\Files\Node;
+use OCP\Files\NotFoundException;
+use OCP\IPreview;
+use OCP\IRequest;
+use OCP\IGroupManager;
+use OCP\IUserSession;
+
+class PreviewController extends Controller {
```
[the remark this answers is not in the document]
Comment (artonge) [reply lib/Controller/PreviewController.php:44]: `OC\Core\Controller\PreviewController` is not public

File: src/components/Albums/CollaboratorsSelectionForm.vue [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+	},
+
+	methods: {
+		/**
+		 * Fetch possible collaborators.
+		 */
+		async searchCollaborators() {
+			try {
+				this.loadingCollaborators = true
+				const response = await axios.get(generateOcsUrl('core/autocomplete/get'), {
```
[the remark this answers is not in the document]
Comment (artonge) [reply src/components/Albums/CollaboratorsSelectionForm.vue:236]: What about `enable_share_accept` ?

[the remark this answers is not in the document]
Comment (artonge) [reply src/components/Albums/CollaboratorsSelectionForm.vue:236]: - If the `files_sharing` app is not enabled, I hide any buttons showing to `CollaboratorsSelectionForm`
- I added support to `sharing.minSearchStringLength` in `CollaboratorsSelectionForm`

File: src/services/PhotoSearch.js [hunk 3]
```diff
@@ -147,7 +147,7 @@ export default async function(path = '', options = {}) {
 
 	return response.data
 		.map(data => genFileInfo(data))
-		// remove prefix path from full file path
-		.map(data => ({ ...data, filename: data.filename.replace(prefixPath, '') }))
+	// remove prefix path from full file path
+	// .map(data => ({ ...data, filename: data.filename.replace(prefixPath, '') }))
```
[the remark this answers is not in the document]
Comment (artonge) [reply src/services/PhotoSearch.js:151]: Cleaned

File: lib/Controller/PreviewController.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (skjnldsv) [reply lib/Controller/PreviewController.php]: Well, I think we should do it anyway :)

**Review by PVince81 — APPROVED** [review]
Reviewed together, looks good :+1:

the only critical part I think is the DB schema to remove the cached display name and use the display name cache instead / lazy user

**Comment by PVince81** [comment]
for retrieving display names, use `$this->userManager->getDisplayName($userId)` to make sure it hits the cache

File: lib/Album/AlbumMapper.php [hunk 4]
```diff
@@ -220,4 +246,179 @@ private function getLastAdded(int $albumId): int {
...
+					throw new \Exception('Invalid collaborator source: ' . $row['collaborator_source']);
+			}
+
+			if (is_null($collaborator)) {
+				return null;
+			}
+
+			return [
+				'id' => $row['collaborator_id'],
+				'label' => $collaborator->getDisplayName(),
```
[the remark this answers is not in the document]
Comment (artonge) [reply lib/Album/AlbumMapper.php:283]: Actually, I don't cache the displayName in the DB

**Pull request merged**